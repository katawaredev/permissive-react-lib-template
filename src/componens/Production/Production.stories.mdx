import { Meta, Story, ArgsTable, Preview } from "@storybook/addon-docs/blocks";
import Production from "./Production";

<Meta title="Production" component={Production} />

# Production

## Preview

export const Template = (args) => <Production {...args} />;

<Preview>
  <Story name="production">{Template.bind({})}</Story>
</Preview>

## Props

<ArgsTable of={Production} />
