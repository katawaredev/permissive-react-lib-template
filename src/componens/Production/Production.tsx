import { FC } from "react";

interface ProductionType {
  /** demo description */
  type?: string;
}

const Production: FC<ProductionType> = ({ type = "bundle" }) => (
  <div>This is a production {type}</div>
);

export default Production;
