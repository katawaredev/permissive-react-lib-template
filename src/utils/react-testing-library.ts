import { ReactElement } from "react";
import { RenderOptions, RenderResult, render } from "@testing-library/react";

import AppWrapper from "AppWrapper";

const customRender = (
  ui: ReactElement,
  options?: RenderOptions
): RenderResult => render(ui, { wrapper: AppWrapper, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
