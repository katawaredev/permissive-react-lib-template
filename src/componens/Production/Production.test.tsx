import { render } from "react-testing-library";
import Production from "./Production";

describe("Production component", () => {
  it("matches default snapshot", () => {
    const { asFragment } = render(<Production />, {});
    expect(asFragment()).toMatchSnapshot();
  });
  it("matches custom snapshot", () => {
    const { asFragment } = render(<Production type="custom" />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
