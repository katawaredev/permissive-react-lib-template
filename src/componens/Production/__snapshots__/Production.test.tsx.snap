// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Production component matches custom snapshot 1`] = `
<DocumentFragment>
  <div>
    This is a production custom
  </div>
</DocumentFragment>
`;

exports[`Production component matches default snapshot 1`] = `
<DocumentFragment>
  <div>
    This is a production bundle
  </div>
</DocumentFragment>
`;
